src/journal/bsod.c

```diff
@@ -146,6 +166,13 @@ static int display_emergency_message_fullscreen(const char *message) {
 
         assert(message);
 
+        if (message_id)
+                message_length = strlen(message) + STRLEN(". Message ID: ") + strlen(message_id) + 2;
+        else
+                message_length = strlen(message) + STRLEN(". Message ID: ") + 2;
+
+        char message_with_id[message_length];
```

Comment:
hmm, let's not use the stack for ths.

instead maybe:

```c
/* If we have the message ID suffix it to the message we print */
_cleanup_free char *combined_message = NULL;
if (message_id)  {
        combined_message = strjoin(message, "\nMessage ID: ", message_id);
        if (!combined_message)
                 return log_oom();
        message = combined_message;
}

Reply:
This makes so much more sense. Thank you sir.

Reply:
> hmm, let's not use the stack for ths.
> 
> instead maybe:
> 
> ```c
> /* If we have the message ID suffix it to the message we print */
> _cleanup_free char *combined_message = NULL;
> if (message_id)  {
>         combined_message = strjoin(message, "\nMessage ID: ", message_id);
>         if (!combined_message)
>                  return log_oom();
>         message = combined_message;
> }
> ```

This is done sir.

Reply:
@bluca,
Could you help check why CI fails sir?